server/pulp/server/webservices/views/search.py

```diff
@@ -154,6 +151,12 @@ def _generate_response(cls, query, options, *args, **kwargs):
         if hasattr(cls, 'model'):
             search_method = cls.model.objects.find_by_criteria
         else:
+            # id field should always be in query.fields. We are forced to do this twice, here and
+            # in the queryset for the mongoengine work. This is necessary because in mongoengine
+            # id is an alias to _id, and queries must be translated to reflect that. If we append
+            # id before this split, our serializers will translate this id also.
+            if query.fields and 'id' not in query.fields:
```

Comment:
`in` comes before `or` in the Python the order of precedence, so that won't work.

``` python
if 'a' in None or ['a']:
    print "or first"

>>> TypeError: argument of type 'NoneType' is not iterable


```